src/main/java/teammates/logic/core/FeedbackResponsesLogic.java

```diff
@@ -504,6 +504,23 @@ public void updateFeedbackResponsesForChangingTeam(
             }
         }
     }
+    
+    /**
+     * Updates responses for a team when its name changes. This is done by
+     * changing responses associated with the old team name to be associated
+     * with the new team name.
+     * @param courseId
+     * @param oldTeam
+     * @param newTeam
+     * @throws EntityDoesNotExistException
+     * @throws InvalidParametersException
+     */
+    public void updateFeedbackResponsesForChangingWholeTeam(String courseId, String oldTeam, String newTeam)
+            throws EntityDoesNotExistException, InvalidParametersException {
+        if (studentsLogic.getStudentsForTeam(oldTeam, courseId).isEmpty()) {
```

Comment:
I don't think you should rely on this (`getStudentsForTeam` to tell if the team is empty) if you are able to get this information directly from the form submission. Otherwise this will be susceptible to eventual consistency and may cause the deletion of valid responses?